src/controls/Keyboard.js

```diff
@@ -1,102 +1,37 @@
-    }
+const actionMap = parseActionConfig()
 
-    window.addEventListener('keydown', downHandler, { passive: true })
-    window.addEventListener('keyup', upHandler, { passive: true })
+function useKeys() {
+  useEffect(() => {
+    window.addEventListener('keydown', (e) => downHandler(e, actionMap), { passive: true })
+    window.addEventListener('keyup', (e) => upHandler(e, actionMap), { passive: true })
 
     return () => {
       window.removeEventListener('keydown', downHandler)
       window.removeEventListener('keyup', upHandler)
     }
-  }, [keyConfig])
+  }, [])
 }
 
-export function KeyboardControls() {
-  const set = useStore((state) => state.set)
+function useTouch() {
+  useEffect(() => {
+    window.addEventListener('touchstart', (e) => downHandler(e, actionMap), { passive: true })
+    window.addEventListener('touchend', (e) => upHandler(e, actionMap), { passive: true })
+    return () => {
+      window.removeEventListener('touchstart', downHandler)
+      window.removeEventListener('touchend', upHandler)
+    }
+  }, [])
+}
 
-  useKeys([
-    {
-      keys: ['ArrowUp', 'w', 'W'],
-      fn: (forward) => set((state) => ({ ...state, controls: { ...state.controls, forward } })),
-    },
-    {
-      keys: ['ArrowDown', 's', 'S'],
-      fn: (backward) => set((state) => ({ ...state, controls: { ...state.controls, backward } })),
-    },
-    {
-      keys: ['ArrowLeft', 'a', 'A'],
-      fn: (left) => set((state) => ({ ...state, controls: { ...state.controls, left } })),
-    },
-    {
-      keys: ['ArrowRight', 'd', 'D'],
-      fn: (right) => set((state) => ({ ...state, controls: { ...state.controls, right } })),
-    },
-    {
-      keys: [' '],
-      fn: (brake) => set((state) => ({ ...state, controls: { ...state.controls, brake } })),
-    },
-    {
-      keys: ['h', 'H'],
-      fn: (honk) => set((state) => ({ ...state, controls: { ...state.controls, honk } })),
-    },
-    {
-      keys: ['Shift'],
-      fn: (boost) => set((state) => ({ ...state, controls: { ...state.controls, boost } })),
-    },
-    {
-      keys: ['r', 'R'],
-      fn: (reset) => set((state) => ({ ...state, controls: { ...state.controls, reset } })),
-    },
-    {
-      keys: ['e', 'E'],
-      fn: () => set((state) => ({ ...state, editor: !state.editor })),
-      up: false,
-    },
-    {
-      keys: ['i', 'I'],
-      fn: () => set((state) => ({ ...state, help: !state.help })),
-      up: false,
-    },
-    {
-      keys: ['m', 'M'],
-      fn: () => set((state) => ({ ...state, map: !state.map })),
-      up: false,
-    },
-    {
-      keys: ['c', 'C'],
-      fn: () =>
-        set((state) => {
-          const current = cameras.indexOf(state.camera)
-          const next = (current + 1) % cameras.length
-          const camera = cameras[next]
-          return { ...state, camera }
-        }),
-      up: false,
-    },
-  ])
+export function KeyboardControls() {
+  useKeys()
+  if (isMobile) {
+    useTouch()
+  }
```

Comment:
```suggestion
  if (isMobile) {
    useTouch()
    return
  }
  useKeys()
```

Reply:
I would leave it like this so you can use both the touch and keyboard control on an iPad.